src/CookieStore.ts

```diff
@@ -72,6 +84,8 @@ class CookieStore {
    * Returns a collection of all stored cookies.
    */
   getAll(): Store {
+    this._deleteExpiredCookies()
+
     return this.store
```

Comment:
Perhaps an architectural overhaul, but we may also declare a custom getter function for `store` that would always ensure store has no expired cookies.

```ts
class CookieStore {
  private _store: Map<...>()

  get store() {
    this.deleteExpiredCookies()
    return this._store
  }
}
```

Benefits:

- Makes expired cookies impossible in the store upon reference of any origin.

Drawbacks:

- Requires to keep a private copy of the `_store` to operate on.

Bringing it up only as a thought, there's no need to do anything about this at the moment. I'm curious to know what you think about this.

Reply:
Yes, maybe it's better to turn this into a new issue.

Isn't the store private anyway?

One thing I noted is that we return the internal `Map` instances from the `CookieStore`. Since we are the only consumer of this package that's not a big problem but in general I think it's a bit dangerous to do that.

Reply:
Agree, let's review this in a separate issue.

What kind of data would you expose when retrieving the cookies? Plain object?